spec/reader_spec.rb

```diff
@@ -128,4 +128,11 @@
     end
   end
 
+  describe "#root" do
+    it "raises an exception if trailer Root is not a dict" do
+      filename = pdf_spec_file("trailer_root_is_not_a_dict")
+      pdf = PDF::Reader.new(filename)
+      expect { pdf.send :root }.to raise_error(PDF::Reader::MalformedPDFError)
+    end
+  end
```

Comment:
My strong preference is to not test private methods using `Object#send`.

A spec that tests `root` indirectly via the `metadata` or  `page_count` methods would be ace. An additional spec at the bottom of `spec/integration_spec.rb` would also be good - I rely on those to prevent regressions in the public API when internal refactoring happens.

Reply:
👍 Have changed the test to call `pdf.metadata`, and added another spec to `spec/integration_spec.rb.